pkg/operator/ceph/cluster/controller.go

```diff
@@ -762,7 +762,13 @@ func (c *ClusterController) handleDelete(cluster *cephv1.CephCluster, retryInter
 		// TODO: filter this List operation by cluster namespace on the server side
 		vols, err := c.volumeAttachment.List(operatorNamespace)
 		if err != nil {
-			return fmt.Errorf("failed to get volume attachments for operator namespace %s: %+v", operatorNamespace, err)
+			// If the flex volume driver is disabled, there is no Volume CRD
+			// defined and the List call will fail. Ignore this error in
+			// order to remove the finalizer anyway, since there shouldn't
+			// be any volume attachments to worry about.
+			// TODO: Is it possible to check if the flex volume CRD is defined?
+			logger.Infof("failed to get volume attachments for cluster %s operator namespace %s: %+v", cluster.Namespace, operatorNamespace, err)
```

Comment:
How about if we check the if `ROOK_ENABLE_FLEX_DRIVER` env var is false? It should really be disabled if the VolumeAttachment CRD isn't even enabled.

Reply:
Great, will do that!